nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_prototype.rs

```diff
@@ -567,8 +568,42 @@ impl StringPrototype {
         todo!()
     }
 
-    fn normalize(_agent: &mut Agent, _this_value: Value, _: ArgumentsList) -> JsResult<Value> {
-        todo!()
+    /// ### [22.1.3.15 String.prototype.normalize ( \[ form \] )](https://tc39.es/ecma262/#sec-string.prototype.normalize)
+    fn normalize(
+        agent: &mut Agent,
+        this_value: Value,
+        arguments: ArgumentsList,
+    ) -> JsResult<Value> {
+        // 1. Let O be ? RequireObjectCoercible(this value).
+        let o = require_object_coercible(agent, this_value)?;
+
+        // 2. Let S be ? ToString(O).
+        let s = to_string(agent, o)?;
+
+        // 3. If form is undefined, let f be "NFC".
+        let form = arguments.get(0);
+        let f = if form.is_undefined() {
+            NormalizeForm::NFC
+        } else {
+            // 4. Else, let f be ? ToString(form).
+            let form_result = NormalizeForm::from_str(form.to_string(agent).unwrap().as_str(agent));
```

Comment:
**issue:** The `unwrap()` here is not according to spec: Spec says `? ToString(form)` so we should do `to_string(agent, form)?`.

Reply:
Oh, yeah. That's right! Thanks for catching this!